src/js/src/mixins/HistoryRoutingUtils.js

```diff
@@ -2,10 +2,34 @@ export default {
   methods: {
     $_pushSearchHistory(destination, query, appliedFacets, solrSettings) {
     // Should be deleted before production.
-    console.log('We pushed a history.')
-    let newFacetUrl = appliedFacets !== '' ? '&facets=' + encodeURIComponent(appliedFacets) : ''
-    let newOptionsUrl = '&offset=' + solrSettings.offset + '&grouping=' + solrSettings.grouping + '&imgSearch=' + solrSettings.imgSearch + '&urlSearch=' + solrSettings.urlSearch
-    history.pushState({name: destination}, destination, '?q=' + query + newFacetUrl + newOptionsUrl)
+    //console.log('We about to push a history.', appliedFacets)
+    let currentQuery = this.$router.history.current.query
+    if(query !== '' && query !== undefined && (currentQuery.query !== query ||
+       currentQuery.grouping !== solrSettings.grouping ||
+       currentQuery.urlSearch !== solrSettings.urlSearch ||
+       currentQuery.imgSearch !== solrSettings.imgSearch ||
+       currentQuery.fileSearch !== solrSettings.fileSearch ||
+       currentQuery.facets !== appliedFacets.join('') ||
+       currentQuery.offset !== solrSettings.offset)) {
```

Comment:
Very beefee if statement. My suggestion would be to separate all the _currentQuery_ stuff out in a seperate function. This function could loop through an array of search functions (grouping, imgSearch, fileSearch etc - remember to special case _facets_  and break out of the loop if at the first sign of _true_) and make the comparisons.

Such a function would clean it up a bit and make it reusable later.

Reply:
I've just moved the code out into its own function, perserving the logic as i was. Imho, the logic you're suggesting would be a little more complex, making arrays to match, and since we're "just" talking about 6 matches to be made at the moment, I don't think the payoff is that great. If you're adamant about it, i'll give it a shot however.